src/Data/Bin/Properties.agda

```diff
@@ -215,16 +215,18 @@ toℕ-homo-+ 2[1+ x ] 2[1+ y ] = begin
   toℕ 2[1+ (suc (x + y)) ]           ≡⟨⟩
   2 ℕ.* (1 ℕ.+ (toℕ (suc (x + y))))  ≡⟨ cong ((2 ℕ.*_) ∘ ℕ.suc) (toℕ-suc (x + y)) ⟩
   2 ℕ.* (2 ℕ.+ toℕ (x + y))          ≡⟨ cong ((2 ℕ.*_) ∘ (2 ℕ.+_)) (toℕ-homo-+ x y) ⟩
-  2 ℕ.* (2 ℕ.+ (toℕ x ℕ.+ toℕ y))    ≡⟨ solve 2 (λ m n → con 2 :* (con 2 :+ (m :+ n))  :=
+  2 ℕ.* (2 ℕ.+ (m ℕ.+ n))            ≡⟨ solve 2 (λ m n → con 2 :* (con 2 :+ (m :+ n)) :=
                                           con 2 :* (con 1 :+ m) :+ con 2 :* (con 1 :+ n))
                                           refl (toℕ x) (toℕ y) ⟩
   toℕ 2[1+ x ] ℕ.+ toℕ 2[1+ y ]      ∎
+  where
+  m = toℕ x;  n = toℕ y
 
 toℕ-homo-+ 2[1+ x ] 1+[2 y ] = begin
   toℕ (2[1+ x ] + 1+[2 y ])             ≡⟨⟩
   toℕ (suc 2[1+ (x + y) ])              ≡⟨ toℕ-suc 2[1+ (x + y) ] ⟩
   ℕ.suc (toℕ 2[1+ (x + y) ])            ≡⟨⟩
-  ℕ.suc (2 ℕ.* (ℕ.suc (toℕ (x + y))))   ≡⟨ cong (λ v → ℕ.suc (2 ℕ.* ℕ.suc v)) (toℕ-homo-+ x y) ⟩
+  ℕ.suc (2 ℕ.* (ℕ.suc (toℕ (x + y))))   ≡⟨ cong (ℕ.suc ∘ (2 ℕ.*_) ∘ ℕ.suc) (toℕ-homo-+ x y) ⟩
```

Comment:
Please could you remove this change, it's much less clear what the function is.

Reply:
Done